network/server.go

```diff
@@ -682,6 +682,9 @@ func (s *Server) SubscribeCh() (<-chan *PeerEvent, error) {
 		}
 	})
 	if err != nil {
+		mutex.Lock()
+		closed = true
```

Comment:
Why not refactor closed to be an int64 and use the `atomic` package for updating only this variable?

Locking down the main mutex here is costly and unnecessary.

I'll overlook the insane fact we use a boolean to manage channels - it's indicative of a bigger design / usage flaw

Reply:
Great advice!